library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -247,17 +247,24 @@ internal class TransactionPayloadFragment :
 
             // The body could either be an image, plain text, decoded binary or not decoded binary.
             val responseBitmap = transaction.responseImageBitmap
-            when {
-                type == PayloadType.RESPONSE && responseBitmap != null -> {
-                    val bitmapLuminance = responseBitmap.calculateLuminance()
-                    result.add(TransactionPayloadItem.ImageItem(responseBitmap, bitmapLuminance))
-                }
-                bodyString.isNotBlank() -> bodyString.lines().forEach {
-                    result.add(TransactionPayloadItem.BodyLineItem(SpannableStringBuilder.valueOf(it)))
-                }
-                !isBodyPlainText -> {
+
+            if (type == PayloadType.RESPONSE && responseBitmap != null) {
+                val bitmapLuminance = responseBitmap.calculateLuminance()
+                result.add(TransactionPayloadItem.ImageItem(responseBitmap, bitmapLuminance))
+                return@withContext result
+            }
+
+            if (bodyString.isBlank()) {
+                val text = requireContext().getString(R.string.chucker_body_empty)
+                result.add(TransactionPayloadItem.BodyLineItem(SpannableStringBuilder.valueOf(text)))
+            } else {
+                if (!isBodyPlainText) {
```

Comment:
This needs to be handled in a different way. Otherwise non-blank bodies that are not plain text but decoded from a different format will not show.

Reply:
Yeah, you are right, just checked. Not sure how we can handle it elegantly, since we also need to understand whenever the content is decoded.

Reply:
Ahh, I'm getting it now. There's one thing I didn't consider when refactoring in #543. `isBodyPlainText` was `true` by default before (same for responses in #554), so it breaks here.

I'm not sure how to proceed with it. IMHO setting property that body is plain text when there is no body was a bug but this complicates things because we can't do check like this.

```kotlin
when {
  isImage -> showImage()
  bodyIsNotEmpty -> showBody()
  bodyIsPlainText -> showEmptyText()
  else -> showEncodedText()
}
```

Maybe the best way would be to have `isBodyEncoded` property instead of `isBodyPlainText`? This way combination of conditions should work properly I think.

Reply:
`isBodyEncoded` would work, I believe.

Reply:
Maybe it is something for a follow up PR?

Reply:
I don't mind fixing it in another PR but wouldn't this make this PR just trade one bug for another?

Reply:
I might misunderstood you initially, because I thought that you want to deal with it.

Reply:
I don't mind dealing with it. I just want get it fixed one way or the other. :)

Reply:
Ok, merging this one. Let's discuss that other fix in Slack.